### DISCUSSION TITLE
GPT Engineer Github integration

### DISCUSSION THREAD

**Discussion by eduzol** [AntonOsika/gpt-engineer#166] [Ideas]
I propose GPT Engineer  should check in his code on a specified github repo. 
This could be the beginning of a CI CD strategy.

**Comment by patillacode** [comment]
This is an interesting idea, I would see this as an optional behavior and a nice to have rather than a need at this point.

**Reply by eduzol** [reply]
True, I tried the project for the first time last night, still core logic needs to become stable. CI/CD could be a further iteration down the road. 
Will try to contribute more often, cheers

**Comment by brainshower** [comment]
I'll bump this.  It's not just checking in the code generated, but using the code generated (previously, or that exists already in the repo) to build upon.  That is, run GPT Engineer at different times, and it uses the existing codebase as a starting point.

**Reply by MMAcode** [reply]
Yeh, i would like to see gpt agent which gets a task, scans existing repo for related code and then writes the code utilising existing code. Is that part of this project? (On a tangent, does this repo have something like architecture designs or sth like overview for developers to see what is already implemented, what's in development and what's on road map?)

**Comment by dlamon1** [comment]
I would pay for this. Anything devops related seems like moving a mountain just to take a step. Not to deviate on the topic here, but as it is related ( somewhat ), the thing on the top of my AI capability wish list is help with deployments/configuration/devops.
